crates/teloxide/src/sugar/bot.rs

```diff
@@ -0,0 +1,410 @@
+    ///
+    /// [`Bot::delete_message`]: crate::Bot::delete_message
+    /// [`Message`]: crate::types::Message
+    fn delete(&self, message: &Message) -> JsonRequest<DeleteMessage>;
+
+    /// This function is the same as [`Bot::copy_message`],
+    /// but can take in [`Message`] to copy it.
+    ///
+    /// [`Bot::copy_messages`]: crate::Bot::copy_message
+    /// [`Message`]: crate::types::Message
+    fn copy<C>(&self, to_chat_id: C, message: &Message) -> JsonRequest<CopyMessage>
+    where
+        C: Into<Recipient>;
+}
+
+impl BotMessagesExt for Bot {
+    fn forward<C>(&self, to_chat_id: C, message: &Message) -> JsonRequest<ForwardMessage>
+    where
+        C: Into<Recipient>,
+    {
+        self.forward_message(to_chat_id, message.chat.id, message.id)
+    }
+
+    fn edit_live_location(
+        &self,
+        message: &Message,
+        latitude: f64,
+        longitude: f64,
+    ) -> JsonRequest<EditMessageLiveLocation> {
+        self.edit_message_live_location(message.chat.id, message.id, latitude, longitude)
+    }
+
+    fn stop_live_location(&self, message: &Message) -> JsonRequest<StopMessageLiveLocation> {
+        self.stop_message_live_location(message.chat.id, message.id)
+    }
+
+    fn set_reaction(&self, message: &Message) -> JsonRequest<SetMessageReaction> {
+        self.set_message_reaction(message.chat.id, message.id)
+    }
+
+    fn pin(&self, message: &Message) -> JsonRequest<PinChatMessage> {
+        self.pin_chat_message(message.chat.id, message.id)
+    }
+
+    fn unpin(&self, message: &Message) -> JsonRequest<UnpinChatMessage> {
+        self.unpin_chat_message(message.chat.id).message_id(message.id)
+    }
+
+    fn edit_text<T>(&self, message: &Message, text: T) -> JsonRequest<EditMessageText>
+    where
+        T: Into<String>,
+    {
+        self.edit_message_text(message.chat.id, message.id, text)
+    }
+
+    fn edit_caption(&self, message: &Message) -> JsonRequest<EditMessageCaption> {
+        self.edit_message_caption(message.chat.id, message.id)
+    }
+
+    fn edit_media(
+        &self,
+        message: &Message,
+        media: InputMedia,
+    ) -> MultipartRequest<EditMessageMedia> {
+        self.edit_message_media(message.chat.id, message.id, media)
+    }
+
+    fn edit_reply_markup(&self, message: &Message) -> JsonRequest<EditMessageReplyMarkup> {
+        self.edit_message_reply_markup(message.chat.id, message.id)
+    }
+
+    fn stop_poll_message(&self, message: &Message) -> JsonRequest<StopPoll> {
+        self.stop_poll(message.chat.id, message.id)
+    }
+
+    fn delete(&self, message: &Message) -> JsonRequest<DeleteMessage> {
+        self.delete_message(message.chat.id, message.id)
+    }
+
+    fn copy<C>(&self, to_chat_id: C, message: &Message) -> JsonRequest<CopyMessage>
+    where
+        C: Into<Recipient>,
+    {
+        self.copy_message(to_chat_id, message.chat.id, message.id)
+    }
+}
+
+#[cfg(test)]
+pub(crate) mod tests {
```

Comment:
I can't quite see how these tests perform anything useful: they seem to just repeat function bodies. Do we really need to test syntax sugar?

Reply:
Ok, fair, these tests probably don't do anything useful